src/Features/Core/Portable/SplitOrMergeIfStatements/Consecutive/AbstractMergeConsecutiveIfStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,197 @@
+    {
+        // Converts:
+        //    if (a)
+        //        Console.WriteLine();
+        //    else if (b)
+        //        Console.WriteLine();
+        //
+        // To:
+        //    if (a || b)
+        //        Console.WriteLine();
+
+        // Converts:
+        //    if (a)
+        //        return;
+        //    if (b)
+        //        return;
+        //
+        // To:
+        //    if (a || b)
+        //        return;
+
+        protected sealed override CodeAction CreateCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument, string ifKeywordText)
+            => new MyCodeAction(createChangedDocument, ifKeywordText);
+
+        protected sealed override async Task<bool> CanBeMergedAsync(
+            Document document, SyntaxNode ifLikeStatement, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var ifGenerator = document.GetLanguageService<IIfLikeStatementGenerator>();
+
+            return CanBeMergedWithParent(syntaxFacts, ifGenerator, ifLikeStatement) ||
+                   await CanBeMergedWithPreviousStatementAsync(document, syntaxFacts, ifGenerator, ifLikeStatement, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected sealed override SyntaxNode GetChangedRoot(Document document, SyntaxNode root, SyntaxNode ifLikeStatement)
+        {
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var ifGenerator = document.GetLanguageService<IIfLikeStatementGenerator>();
+            var generator = document.GetLanguageService<SyntaxGenerator>();
+
+            var isElseIfClause = ifGenerator.IsElseIfClause(ifLikeStatement, out var parentIfLikeStatement);
+            var previousIfLikeStatement = isElseIfClause ? parentIfLikeStatement : GetPreviousStatement(syntaxFacts, ifLikeStatement);
+
+            var newCondition = generator.LogicalOrExpression(
+                ifGenerator.GetCondition(previousIfLikeStatement),
+                ifGenerator.GetCondition(ifLikeStatement));
+
+            newCondition = newCondition.WithAdditionalAnnotations(Formatter.Annotation);
+
+            root = root.TrackNodes(previousIfLikeStatement, ifLikeStatement);
+            root = root.ReplaceNode(
+                root.GetCurrentNode(previousIfLikeStatement),
+                ifGenerator.WithCondition(root.GetCurrentNode(previousIfLikeStatement), newCondition));
+
+            var editor = new SyntaxEditor(root, generator);
+
+            if (isElseIfClause)
+            {
+                // We have:
+                //    if (a)
+                //        Console.WriteLine();
+                //    else if (b)
+                //        Console.WriteLine();
+
+                // Remove the else-if clause and preserve any subsequent clauses.
+
+                ifGenerator.RemoveElseIfClause(editor, root.GetCurrentNode(ifLikeStatement));
```

Comment:
i feel like i've missed something.  where is the code that validated that that the child statements wre the same? (or maybe that's just below this method).

Reply:
above is a method called `CanBeMergedAsync`, which calls `CanBeMergedWithParent` and `CanBeMergedWithPreviousStatementAsync` below, which in turn call `ContainEquivalentStatements`